modules/extensions-support/src/test/java/org/mule/module/extension/HeisenbergOperations.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.api.NestedProcessor;
+import org.mule.api.transport.PropertyScope;
+import org.mule.extension.ExtensionManager;
+import org.mule.extension.annotations.ImplementationOf;
+import org.mule.extension.annotations.Operation;
+import org.mule.extension.annotations.ParameterGroup;
+import org.mule.extension.annotations.RestrictedTo;
+import org.mule.extension.annotations.param.Optional;
+import org.mule.extension.annotations.param.Payload;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+@ImplementationOf(HeisenbergExtension.class)
```

Comment:
I found very counterintuitive to use @ImplementationOf(extension class) to indicate that a class provides operations for that extension